tests/unittests/tests-ipv6_netif/tests-ipv6_netif.c

```diff
@@ -0,0 +1,525 @@
+}
+
+static void test_ipv6_netif_get__empty(void)
+{
+    for (kernel_pid_t pid = KERNEL_PID_FIRST; pid <= KERNEL_PID_LAST; pid++) {
+        TEST_ASSERT_NULL(ng_ipv6_netif_get(pid));
+    }
+}
+
+static void test_ipv6_netif_add_addr__no_iface1(void)
+{
+    ng_ipv6_addr_t addr = {{
+            0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+            0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x00
+        }
+    };
+
+    TEST_ASSERT_EQUAL_INT(-ENOENT, ng_ipv6_netif_add_addr(TEST_UINT16, &addr,
+                          false));
+}
+
+static void test_ipv6_netif_add_addr__no_iface2(void)
+{
+    TEST_ASSERT_EQUAL_INT(-ENOENT, ng_ipv6_netif_add_addr(TEST_UINT16, NULL,
+                          false));
+}
+
+static void test_ipv6_netif_add_addr__addr_NULL(void)
+{
+    test_ipv6_netif_add__success();
+
+    TEST_ASSERT_EQUAL_INT(-EINVAL, ng_ipv6_netif_add_addr(TEST_UINT16, NULL,
+                          false));
+}
+
+static void test_ipv6_netif_add_addr__addr_unspecified(void)
+{
+    ng_ipv6_addr_t addr = NG_IPV6_ADDR_UNSPECIFIED;
+
+    test_ipv6_netif_add__success();
+
+    TEST_ASSERT_EQUAL_INT(-EINVAL, ng_ipv6_netif_add_addr(TEST_UINT16, &addr,
+                          false));
+}
+
+static void test_ipv6_netif_add_addr__ENOMEM(void)
+{
+    ng_ipv6_addr_t addr = {{
+            0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+            0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x00
+        }
+    };
+    int res = 0, i;
+
+    test_ipv6_netif_add__success();
+
+    for (i = 0; res != -ENOMEM; i++, addr.u8[15]++) {
+        res =  ng_ipv6_netif_add_addr(TEST_UINT16, &addr, false);
+    }
+
+    TEST_ASSERT_EQUAL_INT(NG_IPV6_NETIF_ADDR_NUMOF, i);
+}
+
+static void test_ipv6_netif_add_addr__success(void)
+{
+    ng_ipv6_addr_t addr = {{
+            0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+            0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x00
+        }
+    };
+
+    test_ipv6_netif_add__success();
+
+    TEST_ASSERT_EQUAL_INT(0, ng_ipv6_netif_add_addr(TEST_UINT16, &addr, false));
+}
+
+static void test_ipv6_netif_remove_addr__not_allocated(void)
+{
+    ng_ipv6_addr_t addr = {{
+            0xf0, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+            0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x00
+        }
+    };
+
+    test_ipv6_netif_add_addr__success();
+
+    ng_ipv6_netif_remove_addr(TEST_UINT16, &addr);
+
+    TEST_ASSERT_NULL(ng_ipv6_netif_find_addr_local(TEST_UINT16, &addr));
+    addr.u8[0] = 0x01;
+    TEST_ASSERT_NOT_NULL(ng_ipv6_netif_find_addr_local(TEST_UINT16, &addr));
+}
+
+static void test_ipv6_netif_remove_addr__success(void)
+{
+    ng_ipv6_addr_t addr = {{
+            0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+            0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x00
+        }
+    };
+
+    test_ipv6_netif_add_addr__success();
+
+    ng_ipv6_netif_remove_addr(TEST_UINT16, &addr);
+
+    TEST_ASSERT_NULL(ng_ipv6_netif_find_addr_local(TEST_UINT16, &addr));
```

Comment:
No, in `test_ipv6_netif_remove_addr__not_allocated` `addr.u8[0]` is `0xf0` (so "not allocated"), while we here remove the address allocated in `test_ipv6_netif_add_addr__success()`.